packages/rum-core/src/domain/action/getActionNameFromElement.ts

```diff
@@ -9,41 +9,56 @@ import type { RumConfiguration } from '../configuration'
  */
 export const DEFAULT_PROGRAMMATIC_ACTION_NAME_ATTRIBUTE = 'data-dd-action-name'
 export const ACTION_NAME_PLACEHOLDER = 'Masked Element'
+
+type ActionName = {
+  name: string
+  namingSource: string
+}
 export function getActionNameFromElement(
   element: Element,
   { enablePrivacyForActionName, actionNameAttribute: userProgrammaticAttribute }: RumConfiguration,
   nodePrivacyLevel?: NodePrivacyLevel
-): string {
+): ActionName {
   // Proceed to get the action name in two steps:
   // * first, get the name programmatically, explicitly defined by the user.
-  // * then, use strategies that are known to return good results. Those strategies will be used on
-  //   the element and a few parents, but it's likely that they won't succeed at all.
+  // * then, if privacy is set to mask, return a placeholder for the undefined.
+  // * if privacy is not set to mask, use strategies that are known to return good results.
+  //   Those strategies will be used on the element and a few parents, but it's likely that they won't succeed at all.
   // * if no name is found this way, use strategies returning less accurate names as a fallback.
   //   Those are much likely to succeed.
   const defaultActionName =
     getActionNameFromElementProgrammatically(element, DEFAULT_PROGRAMMATIC_ACTION_NAME_ATTRIBUTE) ||
     (userProgrammaticAttribute && getActionNameFromElementProgrammatically(element, userProgrammaticAttribute))
 
-  if (nodePrivacyLevel === NodePrivacyLevel.MASK) {
-    return defaultActionName || ACTION_NAME_PLACEHOLDER
+  if (defaultActionName) {
+    return { name: defaultActionName, namingSource: 'custom_attribute' }
+  } else if (nodePrivacyLevel === NodePrivacyLevel.MASK) {
+    return { name: ACTION_NAME_PLACEHOLDER, namingSource: 'mask_placeholder' }
   }
 
-  return (
-    defaultActionName ||
-    getActionNameFromElementForStrategies(
-      element,
-      userProgrammaticAttribute,
-      priorityStrategies,
-      enablePrivacyForActionName
-    ) ||
-    getActionNameFromElementForStrategies(
-      element,
-      userProgrammaticAttribute,
-      fallbackStrategies,
-      enablePrivacyForActionName
-    ) ||
-    ''
+  const standardName = getActionNameFromElementForStrategies(
+    element,
+    userProgrammaticAttribute,
+    priorityStrategies,
+    enablePrivacyForActionName
   )
+  if (standardName) {
+    return { name: standardName, namingSource: 'standard_attribute' }
```

Comment:
**💬 suggestion:** ‏`priorityStrategies` don't always return standard attribute values though, sometimes it is `text_content`. Maybe strategies can return an `ActionName` object, that we can use to get both the name and correct source.

(Sorry if this is what you asked confirmation for during the standup, I might have misunderstood)

Reply:
Yes, I have a local version that manually returns different source from each strategy. But I wonder if we could also make `getTextualContent` return `ActionName` object.

Reply:
Returning an ActionName object from getTextualContent sounds great. Maybe rename the function to `getActionNameFromTextContent` or something, it could be more explicit